client/src/utils/misc.js

```diff
@@ -1 +1,5 @@
 export const scrollUp = () => window.scrollTo({top: 0, behavior: 'smooth'});
+export const matchRuleShort = (str, rule) => {
+  const escapeRegex = (str) => str.replace(/([.*+?^=!:${}()|[\]/\\])/g, "\\$1");
+  return new RegExp("^" + rule.split("*").map(escapeRegex).join(".*") + "$").test(str);
+}
```

Comment:
Cross posting the answer from Slack:

Take a look in `client/src/constants/routes.js` and see that there are the two hostnames that determine which app (technically, which router) is used.

- `hcapparticipants.*`
- `hcapemployers.*`

Both of these include `*` wildcards so that our local, dev, test, and prod URLs work.

The `matchRuleShort` function looks like a super convoluted way to essentially match against hostnames with wildcard support. The gist of it is:

1. Escape any regex special characters in the pattern (note the capture group)
2. Replace what we’re using as a wildcard (`*`) with a regex wildcard (`.*`)
3. Match string against pattern

Despite its super complex implementation, it doesn’t even handle `www` prefix :face_with_rolling_eyes: To remove complexity, I’d vote we just trust the patterns supplied to this function. Also replace `^` with `^(?:www)?` to handle subdomain. Finally banish that string concat and use template literals.